src/components/PaginationContainer.ts

```diff
@@ -0,0 +1,150 @@
+import { Component, ReactElement, createElement } from "react";
+import { findDOMNode } from "react-dom";
+import * as dijitRegistry from "dijit/registry";
+import * as classNames from "classnames";
+import * as dojoLang from "dojo/_base/lang";
+import * as dojoConnect from "dojo/_base/connect";
+
+import { Pagination, PaginationProps } from "./Pagination";
+import { ValidateConfigs } from "./ValidateConfigs";
+import {
+    ListView,
+    PaginationContainerProps,
+    PaginationContainerState,
+    WrapperProps,
+    parseStyle
+} from "../utils/ContainerUtils";
+import "../ui/Pagination.css";
+
+export default class PaginationContainer extends Component<PaginationContainerProps, PaginationContainerState> {
+    private navigationHandler: object;
+    private listListViewHeight: number;
+
+    constructor(props: PaginationContainerProps) {
+        super(props);
+
+        this.state = {
+            findingListviewWidget: true,
+            maxPageSize: 0,
+            offSet: 1,
+            showPageButton: true
+        };
+
+        this.updateListView = this.updateListView.bind(this);
+        this.transformListView = this.transformListView.bind(this);
+
+        this.navigationHandler = dojoConnect.connect(
+            props.mxform,
+            "onNavigation",
+            this,
+            dojoLang.hitch(this, this.validateConfigs)
+        );
+    }
+
+    render() {
+        return createElement("div",
+            {
+                className: classNames("widget-pagination", this.props.class),
```

Comment:
I believe this is subject to testing and shouldn't be in the container. Root widget node should not be in the container unless it's unavoidable, like the recent Mx7.6 issue. Even then, I'd set these kinds of classes in the actual component where they can be tested & css added.